rio_toa/reflectance.py

```diff
@@ -84,37 +115,72 @@ def calculate_landsat_reflectance(src_path, src_mtl, dst_path, creation_options,
     """
     mtl = toa_utils._load_mtl(src_mtl)
 
-    M = toa_utils._load_mtl_key(mtl,
-        ['L1_METADATA_FILE', 'RADIOMETRIC_RESCALING', 'REFLECTANCE_MULT_BAND_'],
-        band)
-    A = toa_utils._load_mtl_key(mtl,
-        ['L1_METADATA_FILE', 'RADIOMETRIC_RESCALING', 'REFLECTANCE_ADD_BAND_'],
-        band)
-    E = toa_utils._load_mtl_key(mtl, 
-        ['L1_METADATA_FILE', 'IMAGE_ATTRIBUTES','SUN_ELEVATION'])
+    E = mtl['L1_METADATA_FILE']['IMAGE_ATTRIBUTES']['SUN_ELEVATION']
+    date_collected = mtl['L1_METADATA_FILE']['PRODUCT_METADATA']['DATE_ACQUIRED']
+    time_collected_utc = mtl['L1_METADATA_FILE']['PRODUCT_METADATA']['SCENE_CENTER_TIME']
 
     dst_dtype = np.__dict__[dst_dtype]
+    M, A = [], []
 
-    with rio.open(src_path) as src:
-        dst_profile = src.profile.copy()
+    for band, src_path in zip(bands, src_paths):
+        M.append(toa_utils._load_mtl_key(mtl,
```

Comment:
I know I wrote this function but we should go back to strict dictionary usage, ie:
```
M = [
    toa['L1_METADATA_FILE']['RADIOMETRIC_RESCALING']['REFLECTANCE_MULT_BAND_{}'.format(b)]
        for b in bands
]
```